### PR TITLE
fix: add restricted validation

### LINKED ISSUE #21: Restrict text input to options
Adds a `restricted` property that will either invalidate the input or restrict typing anything other text from options.

### PR DESCRIPTION
**Pull request by megheaiulian**
Fixes #21

### CODE REVIEW

**Review by dotpointer — APPROVED** [review]
Tested both live and the demo, works well. But the tests fail, it should be fixed.

File: paper-autocomplete.js [hunk 1]
```diff
@@ -652,15 +648,17 @@ Polymer({
    * @returns {Boolean}
    */
 	validate() {
-		return this.$.autocompleteInput.validate();
+		const valid = this.$.autocompleteInput.validate();
+		if (this.restricted && valid) {
+			return !(this.$.autocompleteInput.invalid = !this.value);
```
Comment (nomego) [inline paper-autocomplete.js]: Please expand this a bit for clarity.
```suggestion
			const hasValue = this.value != null;
			this.$.autocompleteInput.invalid = !hasValue;
			return hasValue;
```

File: paper-autocomplete.js [hunk 2]
```diff
@@ -652,23 +648,18 @@ Polymer({
...
-
-	/**
-   * Clears the current input
-   */
-	clear() {
-		this._value = '';
-		this._text = '';
-		this._clear();
+		const valid = this.$.autocompleteInput.validate();
+		if (this.restricted && valid) {
```
Comment (cristinecula) [inline paper-autocomplete.js:652]: I don't understand how this works... How does this code ensure that the input value is one of the available options?

Comment (megheaiulian) [reply paper-autocomplete.js:652]: `this.value` will always be `null`, `''` or one of values from options.
`this.value` is set to `''` or `null` when you type anything if it doesn't match.

File: paper-autocomplete.js (hunk 1 above)
Comment (megheaiulian) [reply paper-autocomplete.js]: This could also be empty text or maybe undefined (not sure).

Comment (nomego) [reply paper-autocomplete.js]: Well the check handles undefined, but it's a bit messy if it's empty text since it's "declared" as an Object.

**Pull request merged**